File: CourseManagementSystem.API/ClientApp/src/app/viewmodels/testWithSubmissionVM.ts
import {SubmissionAnswerVM} from './testSubmissionVM';

/**
 * class representing a test with a submission
 */
export class TestWithSubmissionVM {
  /**
   * identifier of the test
   */
  public testId: number;

  /**
   * topic of the test
   */
  public testTopic: string;

  /**
   * identifier of the test submission
   */
  public submissionId: number;

  /**
   * answers in the submission
   */
  public answers: SubmissionAnswerWithCorrectAnswerVM[];

  constructor(testId: number, testTopic: string, submissionId: number, answers: SubmissionAnswerWithCorrectAnswerVM[] = []) {
    this.testId = testId;
    this.testTopic = testTopic;
    this.answers = answers;
    this.submissionId = submissionId;
  }

  /**
   * get default instance of this class
   */
  public static getDefault(): TestWithSubmissionVM {
    return new TestWithSubmissionVM(0, '', 0);
  }
}

/**
 * viewmodel representing an answer in submission including correct answer
 */
export class SubmissionAnswerWithCorrectAnswerVM extends SubmissionAnswerVM {
  /**
   * correct answer to the question
   */
  public correctAnswer: string;

  /**
   * points received for the answer
   */
  public receivedPoints: number;

  /**
   * maximal points for the question
   */
  public maximalPoints: number;

  /**
   * comment to the answer provided by teacher
   */
  public comment: string;

  constructor(questionNumber: number, questionText: string, answerText: string, correctAnswer: string,
              receivedPoints: number, maximalPoints: number, comment: string) {
    super(questionNumber, questionText, answerText);
    this.correctAnswer = correctAnswer;
    this.receivedPoints = receivedPoints;
    this.maximalPoints = maximalPoints;
    this.comment = comment;
  }
}

File: CourseManagementSystem.API/ViewModels/TestWithSubmissionVM.cs
using System.Collections.Generic;

namespace CourseManagementSystem.API.ViewModels
{
    /// <summary>
    /// this class represents viewmodel for test with submission
    /// </summary>
    public class TestWithSubmissionVM
    {
        public TestWithSubmissionVM(int testId, string testTopic, int submissionId, IEnumerable<SubmissionAnswerWithCorrectAnswerVM> answers)
        {
            TestId = testId;
            TestTopic = testTopic;
            Answers = answers;
            SubmissionId = submissionId;
        }

        /// <summary>
        /// id of the test
        /// </summary>
        public int TestId { get; }

        /// <summary>
        /// topic of the test
        /// </summary>
        public string TestTopic { get; }

        /// <summary>
        /// id of the test submission
        /// </summary>
        public int SubmissionId { get; }

        /// <summary>
        /// submitted and correct answers
        /// </summary>
        public IEnumerable<SubmissionAnswerWithCorrectAnswerVM> Answers { get; }
    }

    /// <summary>
    /// this class contains student's answer and the correct answer
    /// </summary>
    public class SubmissionAnswerWithCorrectAnswerVM : SubmissionAnswerVM
    {
        public SubmissionAnswerWithCorrectAnswerVM(int questionNumber, string questionText, string answerText, string correctAnswer, int receivedPoints, int maximalPoints)
            : base(questionNumber, questionText, answerText)
        {
            CorrectAnswer = correctAnswer;
            ReceivedPoints = receivedPoints;
            MaximalPoints = maximalPoints;
        }

        /// <summary>
        /// text of the correct answer
        /// </summary>
        public string CorrectAnswer { get; }

        /// <summary>
        /// points received for the answer
        /// </summary>
        public int ReceivedPoints { get; }

        /// <summary>
        /// maximal obtained points for the question
        /// </summary>
        public int MaximalPoints { get; }
    }
}
